include/swupd.h

```diff
@@ -84,6 +84,7 @@ struct file {
 	int last_change;
 	struct update_stat stat;
 
+	unsigned int fd_valid : 1;
```

Comment:
no need for this, once you `close(fd)`, make the `fd = -1` and use that as `fd_valid`